stellarphot/settings/tests/test_custom_widgets.py

```diff
@@ -550,6 +550,141 @@ def test_weird_sequence_of_no_clicks(self, tmp_path):
         # ...the edit/delete buttons should be displayed
         assert choose_or_make_new._edit_delete_container.layout.display != "none"
 
+    @pytest.mark.parametrize("hideable", [True, False])
+    def test_details_hideable_or_not(self, tmp_path, hideable):
+        # The details should be hideable
+        choose_or_make_new = ChooseOrMakeNew(
+            "camera", details_hideable=hideable, _testing_path=tmp_path
+        )
+
+        # New UI element should be in the big box
+        assert choose_or_make_new._show_details_ui in choose_or_make_new.children
+
+        if hideable:
+            # The "show/hide details" widget should be visible
+            assert choose_or_make_new._show_details_ui.layout.display != "none"
+        else:
+            # The "show/hide details" widget should not be visible
+            assert choose_or_make_new._show_details_ui.layout.display == "none"
+
+        # Set "show details" box to unchecked
+        choose_or_make_new._show_details_ui.value = False
+
+        # If hideable then details should not be displayed
+        if hideable:
+            assert choose_or_make_new._details_box.layout.display == "none"
+        else:
+            assert choose_or_make_new._details_box.layout.display != "none"
+
+        # Settings show_details back to true should display details again,
+        # regardless of hideable
+        choose_or_make_new._show_details_ui.value = True
+        assert choose_or_make_new._details_box.layout.display != "none"
+
+    def test_details_hideable_plays_nicely_with_new_item(self, tmp_path):
+        # The details should be hideable and should play nicely with making a new item
+        # Make an item so that selecting "Make new" will count as a value
+        # change later.
+        self.make_test_camera(tmp_path)
+
+        choose_or_make_new = ChooseOrMakeNew(
+            "camera", details_hideable=True, _testing_path=tmp_path
+        )
+
+        assert choose_or_make_new._details_box.layout.display != "none"
+
+        # Set "show details" box to unchecked so details are hidden
+        choose_or_make_new._show_details_ui.value = False
+
+        # Note whether the "show details" box is checked or not
+        show_state = False
```

Comment:
I'm confused by this, it just seems to create a variable `show_state` which is only used once later to assert `False` ... why create the variable?

Reply:
seemed like a good idea at the time, but I don't remember why. Removing it.